src/mobile/src/ui/routes/entry.js

```diff
@@ -79,6 +86,26 @@ const renderInitialScreen = (initialScreen) => {
     });
 };
 
+const resetIfKeychainIsEmpty = (state) => {
+    doesSaltExistInKeychain().then((exists) => {
+        if (!exists) {
+            purgeStoredState({ storage: persistConfig.storage })
+                .then(() => {
+                    state.dispatch(resetWallet());
+                    // Set the new app version
+                    state.dispatch(
+                        setAppVersions({
+                            version: getVersion(),
+                            buildNumber: getBuildNumber(),
+                        }),
+                    );
+                    return initApp(state);
```

Comment:
This might initialise the app instance twice. 

See https://github.com/iotaledger/trinity-wallet/blob/develop/src/mobile/src/libs/store.js#L5 

The app initialisation is triggered as a callback _after_ the persisted state is restored https://github.com/iotaledger/trinity-wallet/blob/develop/src/mobile/src/libs/store.js#L60

Reply:
I would suggest resetting keychain with the state migration setup.